django/db/backends/postgresql/schema.py

```diff
@@ -174,8 +185,41 @@ def _alter_column_type_sql(
         }
         old_is_auto = old_internal_type in auto_field_types
         new_is_auto = new_internal_type in auto_field_types
-        if new_is_auto and not old_is_auto:
```

Comment:
Do we need to check for `serial` fields before `auto` fields? If not I'd prefer to add the new code on the end of the `if/elif` chain - I think it'd make checking the diff a lot more straightforward.

Reply:
We need to check the `auto -> serial` and `serial -> auto` cases, otherwise the `auto -> non-auto` and `non-auto -> auto` branches will execute.